src/main/java/LottoGameController.java

```diff
@@ -0,0 +1,75 @@
+import domain.Lotto;
+import domain.LottoNumber;
+import domain.LottoNumbers;
+import domain.LottoStatisticData;
+import view.InputView;
+import view.OutputView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LottoGameController {
+    private final InputView inputView;
+    private final OutputView outputView;
+
+    public LottoGameController(InputView inputView, OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+    }
+
+    public void run() {
+
+        try {
+            int amount = inputView.readPurchaseAmount();
+            int lottoCount = amount / Lotto.PRICE;
+
+            int manualCount = inputView.readManualCount();
+
+            if (lottoCount < manualCount) {
+                throw new IllegalArgumentException("내신 금액보다 많은 로또를 구매하려고 합니다.");
+            }
+
+            int automaticCount = lottoCount - manualCount;
+
+
+            List<Lotto> lottos = purchaseLottos(automaticCount, manualCount);
+
+            outputView.printPurchasedCount(automaticCount, manualCount);
+            outputView.printLottosNumbers(lottos);
+
+            LottoNumbers winnerNumbers = inputView.readWinnerNumbers();
+            LottoNumber bonusNumber = inputView.readBonusNumber();
+
+
+            LottoStatisticData lottoStatisticData = new LottoStatisticData(winnerNumbers, bonusNumber, lottos);
+
+            outputView.printStatistic(lottoStatisticData.getMatchStatistic());
+
+            double earnRate = lottoStatisticData.calculateRate(amount);
+            outputView.printEarnRate(earnRate);
+
+
+
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    private List<Lotto> purchaseLottos(int autoCount, int manualCount) {
+        List<Lotto> lottos = new ArrayList<>();
+        for (int i = 0; i < manualCount; i++) {
+            LottoNumbers lottoNumbers = inputView.readManualLottoNumbers(i);
+            lottos.add(new Lotto(lottoNumbers));
+        }
```

Comment:
manualCount 만큼 실행 하는 건 괜찮은데, `readManualLottoNumbers(i);` 경우에 flag(i)가 들어가는게 어색하네요.
`i` 라는 인덱스는 InputView의 코드 없이는 현재 객체 내에서 설명이 안되는 값이에요.
flag를 제거할 수 없을까요?

제거한다면, 아래처럼 `Stream`을 활용할 수도 있을 것 같아요.
```suggestion
        inputView.printReadManualLottoMessage();

        List<Lotto> manualLottos = IntStream.range(0, manualCount)
                .mapToObj(i -> inputView.readManualLottoNumbers())
                .map(Lotto::new)
                .toList();
```

Reply:
말씀 하신데로 외부에서 보았을때 flag값의 존재 이유를 모호하게 만드는거 같아서 제거 했습니다!
Stream으로 변경하신걸 보시면, 더 읽기 좋아졌나요?
특정 값을 변환해주는 Map의 함수 등에서는 비교적 읽기가 쉬워진거 같습니다!